commit f755fd5657b619fd27160ad202fc5d773d096e9c
Author: Kay Sievers <kay.sievers@vrfy.org>
Date:   Sat Aug 2 10:26:48 2008 +0200

    rules: run vol_id on opticals only if media is found
    
    Opening an optical drive device node without O_NONBLOCK autocloses the
    tray, we run vol_id on every media change by kernel emitted "change"
    events, which can make it hard to change the media when the tray closes
    immediatey again.:) We check for cdrom_id to indicate an existing track,
    if no media is found, we will not open the device with vol_id.
    
    Thanks to Christian Krause and DavidZ for debugging and testing.

diff --git a/etc/udev/rules.d/60-persistent-storage.rules b/etc/udev/rules.d/60-persistent-storage.rules
index 5ae0c7f..097e864 100644
--- a/etc/udev/rules.d/60-persistent-storage.rules
+++ b/etc/udev/rules.d/60-persistent-storage.rules
@@ -52,6 +52,8 @@ ENV{DEVTYPE}=="partition", ENV{ID_PATH}=="?*", SYMLINK+="disk/by-path/$env{ID_PA
 
 # skip unpartitioned removable media devices from drivers which do not send "change" events
 ENV{DEVTYPE}=="disk", KERNEL!="sd*|sr*", ATTR{removable}=="1", GOTO="persistent_storage_end"
+# skip optical drives without media
+ENV{DEVTYPE}=="disk", KERNEL=="sr*", ENV{ID_CDROM_MEDIA_TRACK_COUNT}!="?*", GOTO="persistent_storage_end"
 
 # import filesystem metadata
 IMPORT{program}="vol_id --export $tempnode"
